doc/whats_new/v1.2.rst

```diff
@@ -101,6 +101,10 @@ Changelog
   deterministic SVD used by the randomized SVD algorithm.
   :pr:`20617` by :user:`Srinath Kailasa <skailasa>`
 
+- |Enhancement| :func:`utils.assert_all_finite` now significantly faster for
```

Comment:
`_assert_all_finite` is a private method, so its name should not be mentioned explicitly.

Either document the impact public utilities such as `sklearn.utils.check_array` and `sklearn.utils.check_X_y` which are calling `_assert_all_finite` internally or alternatively add a generic changelog entry at the top of the changelog, before the list of module because this optimization impacts the input validation of all estimators in scikit-learn, for instance:

```
- |Enhancement| the finiteness checks (detection of NaN and infinite values) in all
  estimators are now significantly more efficient for float32 data by leveraging
  NumPy's SIMD optimized primitives.
  :pr:`23446` by :user:`Meekail Zain <micky774>`
```

```